packages/x-charts/src/hooks/useAxis.ts

```diff
@@ -58,37 +68,84 @@ export function useYAxes() {
 }
 
 /**
- * Get the X axis.
- * @param {AxisId | undefined} axisId - If provided returns the x axis with axisId, else returns the values for the default x axis.
- * @returns The X axis.
+ * Get a specific X axis or the default X axis.
+ *
+ * @param {AxisId} [axisId] - The axis identifier. Can be:
+ *   - A string matching the axis ID defined in the chart's `xAxis` prop
+ *   - A number representing the index in the xAxis array (e.g., 0 for first axis)
+ *   - Undefined to get the default (first) X axis
+ * @returns The configuration for a single X axis.
+ *
+ * @example
+ * // Get the default X axis
+ * const xAxis = useXAxis();
+ *
+ * @example
+ * // Get a specific X axis by string ID
+ * const xAxis = useXAxis('revenue');
+ *
+ * @example
+ * // Get the second X axis by index
+ * const xAxis = useXAxis(1);
  */
 export function useXAxis<T extends keyof AxisScaleConfig>(
   axisId?: AxisId,
 ): ComputedAxis<T, any, ChartsXAxisProps> {
   const store = useStore<[UseChartCartesianAxisSignature]>();
   const { axis: xAxis, axisIds: xAxisIds } = useSelector(store, selectorChartXAxis);
 
-  const id = axisId ?? xAxisIds[0];
+  const id = typeof axisId === 'string' ? axisId : xAxisIds[axisId ?? 0];
```

Comment:
Why are we doing this? A number is a valid axis ID, so this change makes it impossible to access axis whose ID is a number, right? Or am I missing something?